### PR TITLE
Add Menu Overlay

### PR DESCRIPTION
**Pull request by chrismekim**
### Description

- Adds menu overlay

### Type(s) of changes

- [x] New feature
- [x] Update to an existing feature

### How Has This Been Tested?

Click on the menu button in the nav to view.

### Applicable screenshots:

image
image

### Follow-up PR

- Text is overflowing some of the columns on larger screens. I want to see what the actual fonts will look like first since the actual font is more narrow. This will probably be adjusted during QA.›

### CODE REVIEW

**Comment by chrismekim** [comment]
@limsohee1002 @mokaymokay The text in the menu overflows on certain breakpoints (larger screens). The actual fonts are not in and they are much narrower so that will help once I get the fonts in, but the font sizes will probably need to be reworked with XXIX during QA.

File: src/components/MenuOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+              )}
+              to={link.link}
+              ariaLabel={Language.t('Global.generalButtonAriaLabel', {
+                link: link.label,
+              })}
+            >
+              {link.label}
+            </Button>
+          ))}
+        </div>
```
Comment (mokaymokay) [inline src/components/MenuOverlay.tsx:89]: missing some keys

File: src/styles/components/MenuOverlay.scss [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    &--style-small {
+      &::before {
+        content: '';
+        @extend .transition-shorter;
+        background-image: url('../../components/icons/DefaultHeadingLineIcon.svg');
+        display: inline-block;
+        width: 0;
+        opacity: 0;
+      }
```
Comment (mokaymokay) [inline src/styles/components/MenuOverlay.scss:68]: can we make this line style extendable and place in globals or something? seems like it gets applied to different elements

**Review by mokaymokay — APPROVED** [review]

File: src/components/MenuOverlay.tsx (hunk 1 above)
Comment (chrismekim) [reply src/components/MenuOverlay.tsx:89]: !!!! thanks !!

File: src/styles/components/MenuOverlay.scss (hunk 2 above)
Comment (chrismekim) [reply src/styles/components/MenuOverlay.scss:68]: This one has slightly different styling that is only used on the menu overlay :(

**Pull request merged**